### PR TITLE
Expose tritonserver args in values.yaml

### PR DESCRIPTION
**Pull request by okyspace**
Current k8s_onprem chart has predefined args in deployment.yaml. We could change it from the yaml file though but would be cleaner to customise the args through values.yaml instead.

### CODE REVIEW

**Comment by rmccorm4** [comment]
Seems reasonable to me. CC @nealvaidya for comments

**Comment by nealvaidya** [comment]
Looks good to me

File: deploy/k8s-onprem/templates/deployment.yaml [hunk 1]
```diff
@@ -69,7 +69,11 @@ spec:
             limits:
               nvidia.com/gpu: {{ .Values.image.numGpus }}
 
-          args: ["tritonserver", "--model-repository=/models"]
+          args: 
+            - tritonserver
+            {{- range .Values.additionalArgs }}
```
Comment (rmccorm4) [inline deploy/k8s-onprem/templates/deployment.yaml:74]:
```suggestion
            {{- range .Values.serverArgs }}
```

File: deploy/k8s-onprem/values.yaml [hunk 2]
```diff
@@ -35,6 +35,10 @@ image:
   modelRepositoryPath: /srv/models
   numGpus: 1
 
+# add additional args here e.g. --grpc-use-ssl, --grpc-server-certs, repository-poll-secs, etc
+additionalArgs:
```
Comment (rmccorm4) [inline deploy/k8s-onprem/values.yaml:39]:
```suggestion
serverArgs: 
```

**Review by rmccorm4 — CHANGES_REQUESTED** [review]
Hi @okyspace, LGTM overall, just a few things:

1. Rename additionalArgs to serverArgs to be more clear.
2. Can you also share an example command `... --set serverArgs=...` and the output showing that server deployed successfully as a sanity check?
3. Have you signed a CLA already? If not, please follow the instructions [here](https://github.com/triton-inference-server/server/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) and let us know when you've sent it.

**Comment by okyspace** [comment]
@rmccorm4 , Thanks for your reply. 
1. I have updated the naming based on your suggestion. 
2. I can work out an example. Can advise where to provide this example?
3. I will follow up with the CLA. 

Thanks.

**Comment by okyspace** [comment]
@rmccorm4, as requested, I have added the examples for sanity check. Any comments?

-------------------------------------------
The examples show how you can set the customised serverArgs and verify that they are taking effect. 

1. Execute the command below to generate the helm template to check that the serverArgs is being set correctly.  
`helm template  --set serverArgs="{'--model-repository=s3://192.168.1.110:9000/triton-models', '--repository-poll-secs=8', '--model-control-mode=poll', '--grpc-use-ssl=false'}"  triton -n triton .`

You should see the generated **deployment.yaml**. The serverArgs are populated as array args in this example. 
image



2. Run the container using the command. 
`helm install --set serverArgs="{'--model-repository=s3://192.168.1.110:9000/triton-models', '--repository-poll-secs=8', '--model-control-mode=poll', '--grpc-use-ssl=false'}" triton -n triton .`

Run `kubectl describe pods  -n triton`. You should see the args in the description. 
image


Once the container is successfully running, run ```kubectl logs <container id> -n triton```.
You should see these outputs that indicates the args set for the triton server. 

for e.g. model-repository, model-control-mode
image

for e.g. repository-poll-secs. You can see from the logs that "Polling model repository" happens every 8 secs.
image

**Comment by okyspace** [comment]
Hi, is there anything that I need to follow up to close this PR?

File: deploy/k8s-onprem/templates/deployment.yaml [hunk 3]
```diff
@@ -69,7 +69,11 @@ spec:
             limits:
               nvidia.com/gpu: {{ .Values.image.numGpus }}
 
-          args: ["tritonserver", "--model-repository=/models"]
+          args: 
+            - tritonserver
+            {{- range .Values.serverArgs }}
+            - {{.}}
```
Comment (yeahdongcn) [inline deploy/k8s-onprem/templates/deployment.yaml:75]: Nit: `{{.}}` => `{{ . }}`

Comment (okyspace) [reply deploy/k8s-onprem/templates/deployment.yaml]: Thanks. Updated in my latest commit.

**Comment by okyspace** [comment]
@rmccorm4 , is there something i missed out for this PR?

**Review by rmccorm4 — DISMISSED** [review]
This LGTM, thanks for your patience @okyspace :bow:  Letting the linters/pre-commit run to double check.

File: deploy/k8s-onprem/templates/deployment.yaml [hunk 4]
```diff
@@ -69,7 +69,11 @@ spec:
             limits:
               nvidia.com/gpu: {{ .Values.image.numGpus }}
 
-          args: ["tritonserver", "--model-repository=/models"]
+          args:
```
Comment (rmccorm4) [inline deploy/k8s-onprem/templates/deployment.yaml:72]:
```suggestion
          args:
```

File: deploy/k8s-onprem/values.yaml [hunk 5]
```diff
@@ -35,6 +35,10 @@ image:
   modelRepositoryPath: /srv/models
   numGpus: 1
 
+# add server args here e.g. --grpc-use-ssl, --grpc-server-certs, repository-poll-secs, etc
+serverArgs:
```
Comment (rmccorm4) [inline deploy/k8s-onprem/values.yaml:39]:
```suggestion
serverArgs:
```

**Review by rmccorm4 — COMMENTED** [review]
Added suggested fixes for linter errors

**Comment by okyspace** [comment]
@rmccorm4 Thanks for the reviews and suggestions.

**Review by rmccorm4 — APPROVED** [review]

**Comment by rmccorm4** [comment]
> I have also emailed my signed CLA.

Hi @okyspace, can you either resend this email or clarify the name/company attached to it?

**Comment by okyspace** [comment]
@rmccorm4 
I have resend the CLA. 
Name: Ong Kai Yong
Company: N/A. I am not submitting on behalf of a company.

**Comment by rmccorm4** [comment]
Thanks for your contribution and patience @okyspace !

**Pull request merged**